### PR TITLE
Deprecate all old classes from ``optparse``

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.
- [x] If you used multiple emails or multiple names when contributing, add your mails
   and preferred name in ``script/.contributors_aliases.json``

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |

## Description

🎉 

Bit premature because `pyreverse` is still using these, but we might as well deprecate them.

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -47,6 +47,30 @@ Release date: TBA
 
   Ref #5392
 
+* ``UnsupportedAction`` has been deprecated.
```
Comment (Pierre-Sassoulas) [inline ChangeLog:50]: Should we specify that it's because of the optparse migration and what the replacement is ?

File: doc/whatsnew/2.14.rst [hunk 2]
```diff
@@ -97,6 +97,30 @@ Other Changes
 
   Ref #5392
 
+* ``UnsupportedAction`` has been deprecated.
+
+  Ref #5392
+
+* ``OptionsManagerMixIn`` has been deprecated.
```
Comment (Pierre-Sassoulas) [inline doc/whatsnew/2.14.rst:104]: This is highly satisfactory. I'm not going to comment on each mixin but I really hated those classes 😄

Comment (DanielNoord) [reply doc/whatsnew/2.14.rst:104]: Me too! `mypy` despises them too!

File: ChangeLog (hunk 1 above)
Comment (DanielNoord) [reply ChangeLog:55]: Uhm, there is no replacement for most of these. We mention `ArgumentsProvider` and `ArgumentsManager` in another line already. So I think this is fine as is.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**